packages/nimble-components/src/table/models/table-layout-manager.ts

```diff
@@ -0,0 +1,255 @@
+import type { TableRecord } from '../types';
+
+/**
+ * This class manages the layout of columns within a Table.
+ * @interal
+ */
+export class TableLayoutManager<TData extends TableRecord> {
+    @observable
+    public isColumnBeingSized = false;
+
+    @observable
+    public activeColumnIndex?: number;
+
+    private activeColumnDivider?: number;
+    private gridSizedColumns?: TableColumn[];
+    private visibleColumns: TableColumn[] = [];
+    private initialTableScrollableWidth?: number;
+    private initialTableScrollableMinWidth?: number;
+    private initialColumnTotalWidth?: number;
+    private currentTotalDelta = 0;
+    private dragStart = 0;
+    private leftColumnIndex?: number;
+    private rightColumnIndex?: number;
+    private initialColumnPixelWidths: {
+        initalColumnFractionalWidth: number,
+        initialPixelWidth: number,
+        minPixelWidth: number
+    }[] = [];
+
+    public constructor(private readonly table: Table<TData>) {}
+
+    public getGridTemplateColumns(): string {
+        return this.getVisibleColumns()
+            .map(column => {
+                const {
+                    minPixelWidth,
+                    currentPixelWidth,
+                    currentFractionalWidth
+                } = column.columnInternals;
+                if (currentPixelWidth) {
+                    const coercedPixelWidth = Math.max(
+                        minPixelWidth,
+                        currentPixelWidth
+                    );
+                    return `${coercedPixelWidth}px`;
+                }
+
+                return `minmax(${minPixelWidth}px, ${currentFractionalWidth}fr)`;
+            })
+            .join(' ');
+    }
+
+    /**
+     * Sets up state related to interactively sizing a column.
+     * @param dragStart The x-position from which a column size was started
+     * @param activeColumnDivider The divider that was clicked on
+     */
+    public beginColumnInteractiveSize(
+        dragStart: number,
+        activeColumnDivider: number
+    ): void {
+        this.activeColumnDivider = activeColumnDivider;
+        this.leftColumnIndex = this.getLeftColumnIndexFromDivider(
+            this.activeColumnDivider
+        );
+        this.rightColumnIndex = this.leftColumnIndex + 1;
+        this.activeColumnIndex = this.leftColumnIndex + (this.activeColumnDivider % 2);
+        this.dragStart = dragStart;
+        this.currentTotalDelta = 0;
+        this.visibleColumns = this.getVisibleColumns();
+        this.setColumnsToFixedSize();
+        this.initialTableScrollableWidth = this.table.viewport.scrollWidth;
+        this.initialTableScrollableMinWidth = this.table.tableScrollableMinWidth;
+        this.initialColumnTotalWidth = this.getTotalColumnFixedWidth();
+        this.isColumnBeingSized = true;
+        document.addEventListener('mousemove', this.onDividerMouseMove);
+        document.addEventListener('mouseup', this.onDividerMouseUp);
```

Comment:
Extraneous context:

In the [WebVI days](https://dev.azure.com/ni/DevCentral/_git/ASW?path=%2FSource%2FVI%2FHtmlControls.Design%2FWeb%2FFramework%2FniUIActivityService.js&_a=contents&version=GBmain) found there are a bunch of extra cases to deal with. mouseup is not guaranteed so should also listen for window blur (and probably other events) and then there is the slew separate touch events which were deprecated. Which was funny because touch events were deprecated and the replacement api pointer events wasn't standard yet 🤣 BUT TODAY...

Recommendation:

We should switch to [pointer events](https://developer.mozilla.org/en-US/docs/Web/API/Pointer_events) which are mostly analogous to mouse events but provide a unified api across mouse, touch, arbitrary "pointer" concepts.

In particular this use-case seems well suited for the [setPointerCapture](https://developer.mozilla.org/en-US/docs/Web/API/Element/setPointerCapture) workflow which will intercept all other events for us, guarantee the life cycle completes, and actually might obviate the need for the pane of glass (if true, sorry it didn't occur to me sooner 😵)

Reply:
Agreed. I've filed [this issue](https://github.com/ni/nimble/issues/1362) for investigating the usage of pointer events.